sdk/src/main/java/com/silanis/esl/sdk/examples/AttachmentFileExample.java

```diff
@@ -0,0 +1,93 @@
+    public static final String DESCRIPTION1 = "Please upload a scanned copy of your driver's license.";
+    public static final String SIGNER1_ID = "signer1Id";
+    private String attachment1Id;
+    private List<AttachmentRequirement> signer1Attachments;
+    private AttachmentRequirement signer1Att1;
+    public static final String ATTACHMENT_FILE_NAME1 = "The attachment1 for signer1.pdf";
+    public List<AttachmentFile> filesAfterUpload;
+    public List<AttachmentFile> filesAfterDelete;
+
+    public static void main(String... args) {
+        new AttachmentFileExample().run();
+    }
+
+    public AttachmentFileExample() {
+        this.attachmentInputStream1 = this.getClass().getClassLoader().getResourceAsStream("document-for-anchor-extraction.pdf");
+    }
+
+    @Override
+    protected void execute() {
+
+        // Signer1 with 1 attachment requirement
+        signer1 = SignerBuilder.newSignerWithEmail(email1)
+                .withFirstName("John")
+                .withLastName("Smith")
+                .withCustomId(SIGNER1_ID)
+                .withAttachmentRequirement(newAttachmentRequirementWithName(NAME1)
+                        .withDescription(DESCRIPTION1)
+                        .isRequiredAttachment()
+                        .build())
+                .build();
+
+
+        DocumentPackage superDuperPackage = newPackageNamed(getPackageName())
+                .describedAs("This is a package created using the eSignLive SDK")
+                .withSigner(signer1)
+                .withDocument(DocumentBuilder.newDocumentWithName("test document")
+                        .fromStream(documentInputStream1, DocumentType.PDF)
+                        .withSignature(SignatureBuilder.signatureFor(email1)
+                                .build())
+                        .build())
+                .build();
+
+        packageId = eslClient.createAndSendPackage(superDuperPackage);
+
+        reloadPackage();
+
+        byte[] attachment1ForSigner1FileContent = new StreamDocumentSource(attachmentInputStream1).content();
+        eslClient.uploadAttachment(packageId, signer1Att1.getId(), ATTACHMENT_FILE_NAME1,
+                attachment1ForSigner1FileContent, SIGNER1_ID);
+
+        reloadPackage();
+
+        filesAfterUpload = signer1Att1.getFiles();
+        AttachmentFile attachmentFile = filesAfterUpload.get(0);
+
+        eslClient.deleteAttachmentFile(packageId, signer1Att1.getId(), attachmentFile.getId(), SIGNER1_ID);
+
+        reloadPackage();
+
+        filesAfterDelete = signer1Att1.getFiles();
+    }
+
+    private void reloadPackage() {
+        retrievedPackage = eslClient.getPackage(packageId);
+        attachment1Id = retrievedPackage.getSigner(email1).getAttachmentRequirement(NAME1).getId();
+        signer1 = retrievedPackage.getSigner(email1);
+        signer1Attachments = retrievedPackage.getSigner(email1).getAttachmentRequirements();
+        signer1Att1 = signer1Attachments.get(0);
+    }
```

Comment:
It is a good idea to extract the method to avoid duplications. However, this is an example for users, so it is more important to improve user readability.
There is no right answer, but in this case, I think it is better to describe what you need in the example than to extract the method. In particular, the method name `reloadPackage` can be more confusing.